src/ATen/native/xpu/sycl/UpSample.h

```diff
@@ -0,0 +1,161 @@
+#pragma once
+#include <ATen/core/TensorAccessor.h>
+#include <ATen/native/xpu/sycl/Atomics.h>
+
+#include <c10/util/ArrayRef.h>
+#include <c10/util/Optional.h>
+#include <c10/util/OptionalArrayRef.h>
+#include <c10/util/SmallVector.h>
+
+#include <math.h>
+
+namespace at {
+namespace native {
+namespace xpu {
+
+inline size_t idx_cl(
+    const size_t n,
+    const size_t h,
+    const size_t w,
+    const size_t c,
+    const size_t height,
+    const size_t width,
+    const size_t channel) {
+  return ((n * height + h) * width + w) * channel + c;
+}
+
+/* TODO: move this to a common place */
+template <typename scalar_t>
+inline scalar_t min(scalar_t a, scalar_t b) {
+  return a < b ? a : b;
+}
+
+template <typename scalar_t>
+inline scalar_t max(scalar_t a, scalar_t b) {
+  return a > b ? a : b;
+}
+
+// NOTE [ Nearest neighbor upsampling kernel implementation ]
```

Comment:
remove this comment, which is CUDA kernel description

Reply:
done